nc/source/component/Collider.cpp

```diff
@@ -7,29 +7,37 @@
 #include "ui/editor/Widgets.h"
 #endif
 
+#ifdef NC_EDITOR_ENABLED
+namespace
+{
+    bool IsUniformScale(const nc::Vector3& scale)
+    {
+        return nc::math::FloatEqual(scale.x, scale.y) && nc::math::FloatEqual(scale.y, scale.z);
+    }
+}
+#endif
+
 namespace nc
 {
     #ifdef NC_EDITOR_ENABLED
-    Collider::Collider(EntityHandle handle, ColliderType type, Vector3 offset, Vector3 scale)
+    Collider::Collider(EntityHandle handle, ColliderInfo info)
         : ComponentBase(handle),
+          m_type{info.type},
```

Comment:
Curious about this: wouldn't the Collider have a ColliderInfo as well? (Rather than decomposed ColliderInfo members?)